out_writeapi_test.go

```diff
@@ -490,6 +490,206 @@ func TestFLBPluginFlushCtx(t *testing.T) {
+				combinedData = append(combinedData, tempData...)
+			}
+
+			message := dynamicpb.NewMessage(md)
+			err := proto.Unmarshal(combinedData, message)
+			if err != nil {
+				return nil, fmt.Errorf("Failed to unmarshal")
+			}
+
+			textField := message.Get(md.Fields().ByJSONName("Text"))
+			timeField := message.Get(md.Fields().ByJSONName("Time"))
+
+			assert.Equal(t, "FOO", textField.String())
+			assert.Equal(t, "000", timeField.String())
+
+			return nil, nil
+		},
+		FinalizeFunc: func(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		FlushRowsFunc: func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+			return 0, nil
+		},
+		StreamNameFunc: func() string {
+			return ""
+		},
+	}
+
+	// returns mock managedstream
+	origFunc := getWriter
+	getWriter = func(client ManagedWriterClient, ctx context.Context, projectID string, opts ...managedwriter.WriterOption) (MWManagedStream, error) {
+		return mockMS, nil
+	}
+	defer func() { getWriter = origFunc }()
+
+	patchSetContext := monkey.Patch(output.FLBPluginSetContext, func(plugin unsafe.Pointer, ctx interface{}) {
+		setID = ctx.(int)
+	})
+	defer patchSetContext.Unpatch()
+
+	patchSetThreshold := setThreshold
+	setThreshold = func(queueSize int) int {
+		return 4
+	}
+	defer func() { setThreshold = patchSetThreshold }()
+
+	// calls init
+	initRes := FLBPluginInit(nil)
+
+	// begin mocking for flush call
+	orgFunc := getFLBPluginContext
+	getFLBPluginContext = func(ctx unsafe.Pointer) int {
+		return setID
+	}
+	defer func() { getFLBPluginContext = orgFunc }()
+
+	origReadyFunc := isReady
+	isReady = func(queueHead *managedwriter.AppendResult) bool {
+		// Response is not ready for test, preserving queue
+		return false
+	}
+	defer func() { isReady = origReadyFunc }()
+
+	// Returns result of append rows
+	origResultFunc := pluginGetResult
+	pluginGetResult = func(queueHead *managedwriter.AppendResult, ctx context.Context) (int64, error) {
+		return -1, nil
+	}
+	defer func() { pluginGetResult = origResultFunc }()
+
+	patchDecoder := monkey.Patch(output.NewDecoder, func(data unsafe.Pointer, length int) *output.FLBDecoder {
+		return nil
+	})
+	defer patchDecoder.Unpatch()
+
+	// sending rows and data
+	var rowSent int = 0
+	var rowCount int = 5
+	patchRecord := monkey.Patch(output.GetRecord, func(dec *output.FLBDecoder) (ret int, ts interface{}, rec map[interface{}]interface{}) {
+		dummyRecord := make(map[interface{}]interface{})
+		if rowSent < rowCount {
+			rowSent++
+			// Represents "FOO" in bytes as the data for the Text field
+			dummyRecord["Text"] = []byte{70, 79, 79}
+			// Represents "000" in bytes as the data for the Time field
+			dummyRecord["Time"] = []byte{48, 48, 48}
+			return 0, nil, dummyRecord
+		}
+		// Reset to prepare for next call to Flush
+		rowSent = 0
+		return 1, nil, nil
+	})
+	defer patchRecord.Unpatch()
+
+	// creates new stream with a mock build stream fucntion
```

Comment:
"Creates new stream with a mock buildStream function to count the number of times we scale up"